providers/base/api/metrics/api_query_handler.go

```diff
@@ -70,7 +74,12 @@ func NewAPIMetrics() APIMetrics {
 		apiHTTPStatusCodePerProvider: prometheus.NewCounterVec(prometheus.CounterOpts{
 			Namespace: oraclemetrics.OracleSubsystem,
 			Name:      "api_http_status_code",
-			Help:      "Number of API provider responses by status code.",
+			Help:      "Number of API provider responses by status code grouped by category (2XX, 3XX, etc.).",
+		}, []string{providermetrics.ProviderLabel, StatusCodeLabel}),
+		apiHTTPStatusCodeExactPerProvider: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Namespace: oraclemetrics.OracleSubsystem,
+			Name:      "api_http_status_code_granular",
+			Help:      "Number of API provider responses by status code with granularity on the exact code.",
```

Comment:
adding a new metric just so things are backwards compatible and no promql wizardy is required to group status codes (since they are strings and would require regex).

Reply:
instead of a new metric, could we just add a new label?

Reply:
status_code and exact_status_code or smth?

Reply:
fixed

Reply:
good shout

Reply:
No, I don't think we should have a new metric at all. We should just have `api_http_status_code` with labels `status_code` and `status_code_exact`. So we just have one metric and you can use the `status_code` label to aggregate across `2XX`, `4XX`, etc. and the `status_code_exact` label to see if we're specifically hitting `404`s or `429`s.

Reply:
tested with existing dashboards and it works with the single metric with the added label